### PR TITLE
Task: Store matches data

### LINKED ISSUE #9: Store matches data into a database file
- store match results in json file

### PR DESCRIPTION
**Pull request by Raspberry-beans**
The json file format has been created using json5 python package

Relates: 9

### CODE REVIEW

File: projects/news_feed/psl_matches.py [hunk 1]
```diff
@@ -52,9 +52,16 @@ def main():
...
-    #printing all fetched info
+    # printing all fetched info
     print(pandas_data_frame.to_string())
 
+    # converting all fetched information into json file format
+    psl_json_data = simplejson.dumps(all_fetched_information)
+
+    # creating a separate json file named as psl_json_data
+    new_file = open('psl_json_data.txt', 'w')
+    new_file.write(psl_json_data)
```
Comment (M-Shan) [inline projects/news_feed/psl_matches.py]: make a print after this with info saying psl matches data has been saved in <path_to_json_file>

e.g. 

`print("PSL matches data saved in {} file", new_file )`

File: projects/news_feed/psl_matches.py [hunk 2]
```diff
@@ -52,9 +52,16 @@ def main():
...
 
-    #printing all fetched info
+    # printing all fetched info
     print(pandas_data_frame.to_string())
 
+    # converting all fetched information into json file format
+    psl_json_data = simplejson.dumps(all_fetched_information)
+
+    # creating a separate json file named as psl_json_data
+    new_file = open('psl_json_data.txt', 'w')
```
Comment (M-Shan) [inline projects/news_feed/psl_matches.py]: save json file in a proper location e.g. `projects/news_feed/data`

also name variables in a meaningful name e.g. `new_file` --> `psl_data_json_file`

**Review by M-Shan — CHANGES_REQUESTED** [review]

File: projects/news_feed/psl_matches.py (hunk 1 above)
Comment (Raspberry-beans) [reply projects/news_feed/psl_matches.py]: done

File: projects/news_feed/psl_matches.py (hunk 2 above)
Comment (Raspberry-beans) [reply projects/news_feed/psl_matches.py]: done

**Comment by Raspberry-beans** [comment]
@M-Shan the pycache file has been continuously generated whenever I run the code. Seems that it hasn't been ignored.

File: projects/news_feed/psl_matches.py [hunk 3]
```diff
@@ -59,8 +59,11 @@ def main():
     psl_json_data = simplejson.dumps(all_fetched_information)
 
     # creating a separate json file named as psl_json_data
-    new_file = open('psl_json_data.txt', 'w')
-    new_file.write(psl_json_data)
+    psl_json_data_file = open('E:/TechCrunch/projects/news_feed/data/psl_json_data.txt', 'w')
```
Comment (M-Shan) [inline projects/news_feed/psl_matches.py]: don't use hard coded path `E:/TechCrunch/projects/news_feed/data/psl_json_data.txt` use python built-in functions to get current working directory and then create path from there.

`os.path.realpath(os.path.join(os.path.dirname(os.path.realpath(__file__)), "data", "psl_json_data.txt"))`

File: projects/news_feed/psl_matches.py [hunk 4]
```diff
@@ -59,8 +59,11 @@ def main():
...
+    psl_json_data_file.write(psl_json_data)
+
+    # printing that the psl data has been saved in psl_json_data file
+    print("PSL matches data saved in psl_json_data file")
```
Comment (M-Shan) [inline projects/news_feed/psl_matches.py]: use `psl_json_data_file` varaible in print 

`print("PSL matches data saved in {} file".format(psl_json_data_file))`

**Review by M-Shan — CHANGES_REQUESTED** [review]

**Comment by M-Shan** [comment]
> @M-Shan the pycache file has been continuously generated whenever I run the code. Seems that it hasn't been ignored.

that's ok it will be generated but will not appear when you do `git status` since its mentioned in `.gitignore`

File: projects/news_feed/psl_matches.py (hunk 4 above)
Comment (Raspberry-beans) [reply projects/news_feed/psl_matches.py]: done

File: projects/news_feed/psl_matches.py (hunk 3 above)
Comment (Raspberry-beans) [reply projects/news_feed/psl_matches.py]: ok I understand the concept. But confused about its usage 
what this whole line represents and where this line should be included

File: conf/requirements.txt [hunk 5]
```diff
@@ -7,3 +7,4 @@ bs4
 pandas
 requests
 lxml
+json5
```
Comment (M-Shan) [inline conf/requirements.txt]: wrong module !!

it shall be `simplejson`

please take care what you are writing here in future !!

File: projects/news_feed/psl_matches.py [hunk 6]
```diff
@@ -52,9 +52,19 @@ def main():
...
 
-    #printing all fetched info
+    # printing all fetched info
     print(pandas_data_frame.to_string())
 
+    # converting all fetched information into json file format
+    psl_json_data = simplejson.dumps(all_fetched_information)
+
+    # creating a separate json file named as psl_json_data
+    psl_json_data_file = open('E:/TechCrunch/projects/news_feed/data/psl_json_data.txt', 'w')
```
Comment (M-Shan) [inline projects/news_feed/psl_matches.py]: also why extension is `.txt` ?

it shall be `.json`

**Review by M-Shan — CHANGES_REQUESTED** [review]

File: projects/news_feed/psl_matches.py (hunk 3 above)
Comment (Raspberry-beans) [reply projects/news_feed/psl_matches.py]: @M-Shan kindly brief here

File: conf/requirements.txt (hunk 5 above)
Comment (Raspberry-beans) [reply conf/requirements.txt]: My bad, actually I was using json5 first to create json file but it was creating some errors. That's why I switched to simplejson

File: projects/news_feed/psl_matches.py (hunk 6 above)
Comment (Raspberry-beans) [reply projects/news_feed/psl_matches.py]: txt extension has been replaced by json

File: projects/news_feed/psl_matches.py (hunk 3 above)
Comment (M-Shan) [reply projects/news_feed/psl_matches.py]: `os.path.dirname(os.path.realpath(__file__)` to get directory of psl_matches.py file 

`os.path.join(os.path.dirname(os.path.realpath(__file__)), "data", "psl_json_data.txt")` is to create a path of data folder and then psl_json_data.json inside data folder

you can print `os.path.join(os.path.dirname(os.path.realpath(__file__)), "data", "psl_json_data.txt")` to check how path looks like

Comment (Raspberry-beans) [reply projects/news_feed/psl_matches.py]: @M-Shan kindly guide in this issue
error

Comment (M-Shan) [reply projects/news_feed/psl_matches.py]: @Raspberry-beans you have to create this folder by 
```
# making data folder
data_folder_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "data")
os.mkdir(data_folder_path)
psl_json_data_file = open(os.path.join(data_folder_path, "psl_2019_data.json"), 'w')
```

Comment (Raspberry-beans) [reply projects/news_feed/psl_matches.py]: done

File: projects/news_feed/utilities.py [hunk 7]
```diff
@@ -125,3 +127,24 @@ def grab_winning_team(class_attribute):
...
+    in a separate data folder
+
+    :param list fetched_information: Contains information of all psl 2019 matches
+
+    :returns: None
+    """
+
+    psl_json_data = simplejson.dumps(fetched_information)
+    data_folder_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "data")
+    os.mkdir(data_folder_path)
```
Comment (M-Shan) [inline projects/news_feed/utilities.py]: make directory only when it doesnot exist
```
if not os.path.exists(data_folder_path):
    os.mkdir(data_folder_path)
```

**Review by M-Shan — CHANGES_REQUESTED** [review]

**Review by M-Shan — APPROVED** [review]

**Pull request merged**